xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -361,8 +363,19 @@ viewers.Comments = Class.create({
 
               if (this.restartNeeded) {
                 this.container.update(response.responseText);
+
+                // If a content is found in submittedcomment that means the submission was not valid.
+                // For instance, the captcha was not accepted.
+                var submittedCommentContainer =  $("submittedcomment");
+                var submittedComment = '';
+                if(submittedCommentContainer) {
+                  submittedComment = submittedCommentContainer.textContent;
```

Comment:
It's actually an input, shouldn't you use its value?

Reply:
textContent is a hidden div, so its content should be accessed with textContent

Reply:
My bad I was sure it was an input type hidden, I actually don't see why it's not.